packages/smart-wallet/src/smartWallet.js

```diff
@@ -60,161 +64,323 @@ const { details: X, quote: q } = assert;
-
-  /**
-   * To ensure every offer ID is unique we require that each is a number greater
-   * than has ever been used. This high water mark is sufficient to track that.
-   *
-   * @type {number}
-   */
-  let lastOfferId = 0;
-
-  /**
-   * Invitation makers yielded by offer results
-   *
-   * @type {MapStore<number, import('./types').RemoteInvitationMakers>}
-   */
-  const offerToInvitationMakers = makeScalarBigMapStore('invitation makers', {
-    durable: true,
-  });
-
-  /** @type {MapStore<Brand, BrandDescriptor>} */
-  const brandDescriptors = makeScalarMapStore();
-
-  /**
-   * What purses have reported on construction and by getCurrentAmountNotifier updates.
-   *
-   * @type {MapStore<RemotePurse, Amount>}
-   */
-  const purseBalances = makeScalarMapStore();
+export const initState = (unique, shared) => {
+  // TODO move to guard
+  // assert.typeof(address, 'string', 'invalid address');
+  // assert(bank, 'missing bank');
+  // assert(invitationIssuer, 'missing invitationIssuer');
+  // assert(invitationBrand, 'missing invitationBrand');
+  // assert(invitationPurse, 'missing invitationPurse');
+  // assert(storageNode, 'missing storageNode');
 
-  // #endregion
-
-  // #region publishing
   // NB: state size must not grow monotonically
   // This is the node that UIs subscribe to for everything they need.
   // e.g. agoric follow :published.wallet.agoric1nqxg4pye30n3trct0hf7dclcwfxz8au84hr3ht
-  const myWalletStorageNode = E(storageNode).makeChildNode(address);
-
-  /** @type {StoredPublishKit<UpdateRecord>} */
-  const updatePublishKit = makeStoredPublishKit(
-    myWalletStorageNode,
-    marshaller,
+  const myWalletStorageNode = E(shared.storageNode).makeChildNode(
+    unique.address,
   );
 
-  /**
-   * @param {RemotePurse} purse
-   * @param {Amount} balance
-   * @param {'init'} [init]
-   */
-  const updateBalance = (purse, balance, init) => {
-    if (init) {
-      purseBalances.init(purse, balance);
-    } else {
-      purseBalances.set(purse, balance);
-    }
-    updatePublishKit.publisher.publish({
-      updated: 'balance',
-      currentAmount: balance,
-    });
+  const preciousState = {
+    // Invitation makers yielded by offer results
+    offerToInvitationMakers: makeScalarBigMapStore('invitation makers', {
+      durable: true,
+    }),
+    // What purses have reported on construction and by getCurrentAmountNotifier updates.
+    purseBalances: makeScalarMapStore(),
+    // Private purses. This assumes one purse per brand, which will be valid in MN-1 but not always.
+    brandPurses: makeScalarBigMapStore('brand purses', { durable: true }),
   };
 
-  // #endregion
+  const nonpreciousState = {
+    brandDescriptors: makeScalarMapStore(),
+    // To ensure every offer ID is unique we require that each is a number greater
+    // than has ever been used. This high water mark is sufficient to track that.
+    lastOfferId: 0,
```

Comment:
does lastOfferId need to survive upgrade?

Reply:
it's a SHOULD not a MUST. if it doesn't survive it's easy to determine from the `offerStatus` update in RPC. I'll make a note of that.